ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -341,4 +346,61 @@ protected void onOverScrolled(int scrollX, int scrollY, boolean clampedX, boolea
 
     super.onOverScrolled(scrollX, scrollY, clampedX, clampedY);
   }
+
+  @Override
+  public void onChildViewAdded(View parent, View child) {
+    setContentView(1 /* expectedChildCount */);
+  }
+
+  @Override
+  public void onChildViewRemoved(View parent, View child) {
+    setContentView(0 /* expectedChildCount */);
+
+  }
+
+  private void setContentView(int expectedChildCount) {
+    if (mContentView != null) {
+      mContentView.removeOnLayoutChangeListener(this);
+    }
+    if (this.getChildCount() != expectedChildCount) {
+      FLog.w(
+              ReactConstants.TAG,
+              String.format("Expected %s children for ReactScrollView, found %s.", expectedChildCount, this.getChildCount()));
+    }
+
+    if (this.getChildCount() > 0) {
+      mContentView = this.getChildAt(0);
+      mContentView.addOnLayoutChangeListener(this);
+    } else { // no children
+      mContentView = null;
+    }
+  }
+
+  /**
+   * Called when a mContentView's layout has changed.
+   */
+  @Override
+  public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
+    this.onContentLayoutChange(v, left, top, right, bottom, oldLeft, oldTop, oldRight, oldBottom);
+  }
+
+  private void onContentLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
```

Comment:
Just inline this above? Or create a inner class to make things more obvious that onLayoutChange refers to the content view